t/40-expand_placeholders.t

```diff
@@ -0,0 +1,124 @@
+#! /usr/bin/perl
+
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+use strict;
+use warnings;
+
+BEGIN {
+    unshift @INC, 'lib';
+}
+
+use FindBin;
+use lib "$FindBin::Bin/lib";
+use Test::More;
+use OpenQA::ExpandPlaceholder;
+
+my $settings = {
+    BUILD_SDK                => '%BUILD_HA%',
+    BETA                     => 1,
+    ISO_MAXSIZE              => '4700372992',
+    SHUTDOWN_NEEDS_AUTH      => 1,
+    HDD_1                    => 'SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed.qcow2',
+    PUBLISH_HDD_1            => 'SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed.qcow2',
+    ANOTHER_JOB              => 'SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed.qcow2',
+    ARCH                     => 'x86_64',
+    BACKEND                  => 'qemu',
+    BUILD                    => '1234',
+    BUILD_SLE                => '%BUILD%',
+    MACHINE                  => '64bit',
+    PATCH                    => 1,
+    UPGRADE                  => 1,
+    ISO                      => 'SLE-%VERSION%-%FLAVOR%-%MACHINE%-Build%BUILD%-Media1.iso',
+    WORKER_CLASS             => 'qemu_x86_64',
+    VERSION                  => '15-SP1',
+    FLAVOR                   => 'Installer-DVD',
+    ADDONURL_SDK             => 'ftp://openqa.suse.de/SLE-%VERSION%-SDK-POOL-%ARCH%-Build%BUILD_SDK%-Media1/',
+    DEPENDENCY_RESOLVER_FLAG => 1,
+    DESKTOP                  => 'textmode',
+    DEV_IMAGE                => 1,
+    HDDSIZEGB                => 50,
+    INSTALLONLY              => 1,
+    PATTERNS                 => 'base,minimal',
+    PUBLISH_PFLASH_VARS =>
+      'SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed-uefi-vars.qcow2',
+    SEPARATE_HOME      => 0,
+    BUILD_HA           => '%BUILD%',
+    BUILD_SES          => '%BUILD%',
+    WORKAROUND_MODULES => 'base,desktop,serverapp,script,sdk',
+};
+
+subtest expand_placeholders => sub {
+    my $str            = OpenQA::ExpandPlaceholder::expand_placeholders($settings);
```

Comment:
Why not calling the variable `$error`? And add a check `is($error, undef, 'no error returned').`.